src/Microsoft.AspNetCore.Rewrite/RewriteOptionsExtensions.cs

```diff
@@ -117,5 +117,36 @@ public static RewriteOptions AddRedirectToHttps(this RewriteOptions options, int
             options.Rules.Add(new RedirectToHttpsRule { StatusCode = statusCode, SSLPort = sslPort });
             return options;
         }
+
+        /// <summary>
+        /// Redirect a request to www subdomain if the incoming request is non-www, with returning a 301
+        /// status code for permanently redirected.
+        /// </summary>
+        /// <param name="options">The <see cref="RewriteOptions"/>.</param>
+        /// <returns></returns>
+        public static RewriteOptions AddRedirectToWwwPermanent(this RewriteOptions options)
+        {
+            return AddRedirectToWww(options, statusCode: StatusCodes.Status301MovedPermanently);
+        }
+
+        /// <summary>
+        /// Redirect a request to www subdomain if the incoming request is non-www
+        /// </summary>
+        /// <param name="options">The <see cref="RewriteOptions"/>.</param>
+        public static RewriteOptions AddRedirectToWww(this RewriteOptions options)
+        {
+            return AddRedirectToWww(options, statusCode: StatusCodes.Status307TemporaryRedirect);
+        }
+
+        /// <summary>
+        /// Redirect a request to www subdomain if the incoming request is non-www
+        /// </summary>
+        /// <param name="options">The <see cref="RewriteOptions"/>.</param>
+        /// <param name="statusCode">The status code to add to the response.</param>
+        public static RewriteOptions AddRedirectToWww(this RewriteOptions options, int statusCode)
+        {
+            options.Rules.Add(new RedirectToWwwRule { StatusCode = statusCode });
```

Comment:
Is there a way to set AllowLocalhost? I think you need to add another extension:
```csharp
 public static RewriteOptions AddRedirectToWww(this RewriteOptions options, int statusCode, bool allowLocalhost)
```

Reply:
If someone really wants to not make an exception for localhost, it can be set by adding the RedirectToWwwRule explicitly. I imagined it would be used so infrequently that I didn't include it in an extension method.

Having said that, a comment or even an optional param for the extension method would show the expected behaviour more though when someone is using it for the first time and wondering why it isn't working in IISExpress. What do you think?

Reply:
An non-exposed option might as well not exist. Add an overload or remove the option.

Reply:
An optional param on all the extension methods or just this one?

Reply:
An additional overload with all options. Optional params are discouraged in our public API.

Reply:
That's fair enough. Just seems like a lot of bloat for something that won't ever be set to false. Maybe keep the localhost logic but just not make it an option? What do you think? I'm happy with either

Reply:
Sure, go ahead and remove the option.